src/sign.py

```diff
@@ -0,0 +1,174 @@
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Script to sign models."""
+
+import argparse
+import logging
+import pathlib
+
+from model_signing import model
+from model_signing.hashing import file
+from model_signing.hashing import memory
+from model_signing.serialization import serialize_by_file
+from model_signing.signature import fake
+from model_signing.signature import key
+from model_signing.signature import pki
+from model_signing.signature import signing
+from model_signing.signature import sigstore
+from model_signing.signing import in_toto
+from model_signing.signing import in_toto_signature
+
+
+log = logging.getLogger(__name__)
+
+
+def _arguments() -> argparse.Namespace:
+    parser = argparse.ArgumentParser("Script to sign models")
+    parser.add_argument(
+        "--model_path",
+        help="path to the model to sign",
+        required=True,
+        type=pathlib.Path,
+        dest="model_path",
+    )
+    parser.add_argument(
+        "--sig_out",
+        help="the output file, it defaults ./signature.json",
+        required=False,
+        type=pathlib.Path,
+        default=pathlib.Path("./signature.json"),
+        dest="sig_out",
+    )
+
+    method_cmd = parser.add_subparsers(required=True, dest="method")
+    # Sigstore
+    method_cmd.add_parser("sigstore")
+    # PKI
+    pki = method_cmd.add_parser("pki")
+    pki.add_argument(
+        "--cert_chain",
+        help="paths to pem encoded certificate files or a single file \
+            containing a chain",
+        required=False,
+        type=list[str],
+        default=[],
+        nargs="+",
+        dest="cert_chain_path",
+    )
+    pki.add_argument(
+        "--signing_cert",
+        help="the pem encoded signing cert",
+        required=True,
+        type=pathlib.Path,
+        dest="signing_cert_path",
+    )
+    pki.add_argument(
+        "--private_key",
+        help="the path to the private key PEM file",
+        required=True,
+        type=pathlib.Path,
+        dest="key_path",
+    )
+    # private key
+    p_key = method_cmd.add_parser("private-key")
+    p_key.add_argument(
+        "--private_key",
+        help="the path to the private key PEM file",
+        required=True,
+        type=pathlib.Path,
+        dest="key_path",
+    )
+    # skip
+    method_cmd.add_parser("skip")
+
+    return parser.parse_args()
+
+
+def _get_payload_signer(args: argparse.Namespace) -> signing.Signer:
+    if args.method == "sigstore":
+        signer = sigstore.SigstoreSigner()
+        log.info(f"Signing ID provider: {signer.id_provider}")
+        log.info(f"Signing ID: {signer.signing_id}")
```

Comment:
Why log these only for sigstore?

Reply:
no logging anymore but for the others it was clear from the input parameters. Only sigstore potentially invokes an interactive process, therefore the logging was meant to reinforce what the user typed in the interactive window.